pkg/apis/management.cattle.io/v3/authz_types.go

```diff
@@ -26,49 +26,95 @@ var (
 // +kubebuilder:skipversion
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// Project is a group of namespaces.
+// Projects are be used to create a multi-tenant environment within a Kubernetes cluster by managing namespace operations,
+// such as role assignments or quotas, as a group.
 type Project struct {
 	types.Namespaced
 
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   ProjectSpec   `json:"spec,omitempty"`
+	// Spec is the specification of the desired configuration for the project.
+	Spec ProjectSpec `json:"spec,omitempty"`
+
+	// Status is the most recently observed status of the project.
 	Status ProjectStatus `json:"status"`
 }
 
 func (p *Project) ObjClusterName() string {
 	return p.Spec.ObjClusterName()
 }
 
+// ProjectStatus represents the most recently observed status of the project.
 type ProjectStatus struct {
-	Conditions                    []ProjectCondition `json:"conditions"`
-	PodSecurityPolicyTemplateName string             `json:"podSecurityPolicyTemplateId"`
-	MonitoringStatus              *MonitoringStatus  `json:"monitoringStatus,omitempty" norman:"nocreate,noupdate"`
+	// Conditions are a set of indicators about aspects of the project.
+	Conditions []ProjectCondition `json:"conditions"`
+
+	// PodSecurityPolicyTemplateName is the pod security policy template associated with the project.
+	PodSecurityPolicyTemplateName string `json:"podSecurityPolicyTemplateId"`
+
+	// MonitoringStatus is the status of the Monitoring V1 app.
+	MonitoringStatus *MonitoringStatus `json:"monitoringStatus,omitempty" norman:"nocreate,noupdate"`
 }
 
+// ProjectCondition is the status of an aspect of the project.
 type ProjectCondition struct {
 	// Type of project condition.
 	Type string `json:"type"`
+
 	// Status of the condition, one of True, False, Unknown.
 	Status v1.ConditionStatus `json:"status"`
+
 	// The last time this condition was updated.
 	LastUpdateTime string `json:"lastUpdateTime,omitempty"`
+
 	// Last time the condition transitioned from one status to another.
 	LastTransitionTime string `json:"lastTransitionTime,omitempty"`
+
 	// The reason for the condition's last transition.
 	Reason string `json:"reason,omitempty"`
+
 	// Human-readable message indicating details about last transition
 	Message string `json:"message,omitempty"`
 }
 
+// ProjectSpec is a description of the project.
 type ProjectSpec struct {
-	DisplayName                   string                  `json:"displayName,omitempty" norman:"required"`
-	Description                   string                  `json:"description"`
-	ClusterName                   string                  `json:"clusterName,omitempty" norman:"required,type=reference[cluster]"`
-	ResourceQuota                 *ProjectResourceQuota   `json:"resourceQuota,omitempty"`
+	// DisplayName is the human-readable name for the project. Required.
```

Comment:
I haven't seen how the output of `kubectl explain` looks for these new resources, so I'm wondering if the indication that a field is required might be duplicated in a few fields here - with the kubebuilder comment and part of the doc comment...